commons-rest-test/src/main/java/com/mule/connectors/commons/rest/test/assertion/body/StatusCodeEquals.java

```diff
@@ -0,0 +1,31 @@
+package com.mule.connectors.commons.rest.test.assertion.body;
+
+import javax.ws.rs.core.Response;
+
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * {@link Matcher} that validates that the status code of a {@link Response} is the expected value.
+ */
+public class StatusCodeEquals extends ResponseBodyAssertion {
```

Comment:
I would make this extend BaseMatcher and implement RequestAndResponseAssertion.
This matcher doesn't apply to the response's body nor it's header, so it shouldn't extend ResponseBodyAssertion.